lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -43,34 +50,28 @@ class BitmapCanvas extends EngineCanvas with SaveStackTracking {
   ///
   /// These pixels are different from the logical CSS pixels. Here a pixel
   /// literally means 1 point with a RGBA color.
-  int get widthInBitmapPixels => _widthInBitmapPixels;
-  int _widthInBitmapPixels;
+  final int _widthInBitmapPixels;
 
   /// The number of pixels along the width of the bitmap that the canvas element
   /// renders into.
   ///
   /// These pixels are different from the logical CSS pixels. Here a pixel
   /// literally means 1 point with a RGBA color.
-  int get heightInBitmapPixels => _heightInBitmapPixels;
-  int _heightInBitmapPixels;
+  final int _heightInBitmapPixels;
 
   /// The number of pixels in the bitmap that the canvas element renders into.
   ///
   /// These pixels are different from the logical CSS pixels. Here a pixel
   /// literally means 1 point with a RGBA color.
-  int get bitmapPixelCount => widthInBitmapPixels * heightInBitmapPixels;
+  int get bitmapPixelCount => _widthInBitmapPixels * _heightInBitmapPixels;
 
   int _saveCount = 0;
 
   /// Keeps track of what device pixel ratio was used when this [BitmapCanvas]
   /// was created.
   final double _devicePixelRatio = html.window.devicePixelRatio;
 
-  // Cached current filter, fill and stroke style to reduce updates to
-  // CanvasRenderingContext2D that are slow even when resetting to null.
-  String _prevFilter = 'none';
-  Object _prevFillStyle;
-  Object _prevStrokeStyle;
+  int _canvasPositionX, _canvasPositionY;
```

Comment:
Like other fields, docs here would be useful.

Reply:
Done.